protocol/src/lib.rs

```diff
@@ -206,14 +209,32 @@ impl FromStr for ModelDescription {
     fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
         let split = s.split("/");
         let tokens: Vec<String> = split.map(|v| v.to_string()).collect();
-        if tokens.len() < 3 || tokens.iter().any(|v| v.is_empty()) {
+        if tokens.len() < 3 || tokens[..2].iter().any(|v| v.is_empty()) {
             return Err(());
         }
+        let maybe_endpoint = if tokens.len() >= 3 {
```

Comment:
nit: should we call this `endpoint`? perhaps we could just initialize this to `None`, and then set it to `Some(...)` within the if blocks, so we can avoid the "else ... None" blocks here:

https://github.com/DJAndries/llmvm/blob/21e50c47652a2cc3f3f1a7acde2cd74d2e37cd7d/protocol/src/lib.rs#L226-L231